authorization/role/model/role_management_model_service.go

```diff
@@ -34,6 +35,121 @@ type GormRoleManagementModelService struct {
 	repository repository.Repositories
 }
 
+// ListByResourceAndRoleName lists role assignments of a specific resource.
+func (r *GormRoleManagementModelService) ListByResourceAndRoleName(ctx context.Context, resourceID string, roleName string) ([]identityrole.IdentityRole, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "identity_role", "list"}, time.Now())
+	var identityRoles []identityrole.IdentityRole
+
+	r.db = r.db.Debug()
+	db := r.db.Raw(`WITH RECURSIVE q AS ( 
+		SELECT 
+		  resource_id, parent_resource_id 
+		FROM 
+		  resource 
+		WHERE 
+		  resource_id = ?
+		UNION ALL
+		SELECT 
+		  p.resource_id, p.parent_resource_id
+		FROM 
+		  resource p
+		JOIN q ON 
+		  q.parent_resource_id = p.resource_id)
+	  SELECT 
+		q.parent_resource_id,q.resource_id, ir.identity_role_id, ir.identity_id, r.role_id, r.name 
+	  FROM 
+		identity_role ir, q, role r
+	  WHERE 
+		ir.resource_id = q.resource_id 
+		and ir.role_id = r.role_id
+		and r.name = ?`, resourceID, roleName)
+
+	rows, err := db.Rows()
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"resource_id": resourceID,
+			"err":         err,
+		}, "error running custom sql to get identity roles")
+		return identityRoles, err
+	}
+	defer rows.Close()
+
+	columns, err := rows.Columns()
+	columnValues := make([]interface{}, len(columns))
+
+	var ignore interface{}
+	for index := range columnValues {
+		columnValues[index] = &ignore
+	}
+
+	if err != nil {
```

Comment:
Move the error check up. Just after `columns, err := rows.Columns()`

Reply:
@alexeykazakov possibly yes, but we can decide later whether we want to return mapped roles also.

Reply:
yeah, guessing just the query would change then.

Reply:
65b4b61